### ISSUE TITLE
TS doesn't detect not defined variables when using ES6 destructuring

### ISSUE THREAD

**Issue by stegano** [microsoft/TypeScript#39801]
**TypeScript Version:**  3.8.3

**Search Terms:**

**Code**

```ts
// index.ts
const [value] = [1, 2, 3].map((v) => {
  console.log(value);
  return v;
});
console.log(value);
```

1) Build to ES6 target using `tsc`
```bash
$ tsc ./index.ts --target es6
```

2) Compiled index.ts file without errors
```js
// index.js
const [value] = [1, 2, 3].map((v) => {
    console.log(value); // value is not defined in this scope
    return v;
});
console.log(value);
```

3) Run the index.js file
```bash
$ node ./index.js

Error occurred as a result of execution
> ReferenceError: value is not defined
...
```

**Expected behavior:**
- An error should occur during build.

**Actual behavior:**
- TS doesn't detect any syntax errors.

**Playground Link:** 

**Related Issues:**

**Comment by RyanCavanaugh** [comment]
Duplicate #11498 - this code *could* be valid because `map` might not immediately invoke its callback

**Comment by stegano** [comment]
@RyanCavanaugh Thanks for your answer
